File: SQL Server 2012/Projects/Visual Basic/Chapter 22/ApClrObjects/Triggers/Vendors_Update_State.vb
Imports System
Imports System.Data
Imports System.Data.SqlClient
Imports System.Data.SqlTypes
Imports Microsoft.SqlServer.Server


Partial Public Class Triggers
    <Microsoft.SqlServer.Server.SqlTrigger(
        Target:="Vendors",
        Event:="AFTER INSERT, UPDATE")> _
    Public Shared Sub Vendors_Update_State()
        ' Open the Connection object for the context
        Dim connection As New SqlConnection("Context connection=true")
        connection.Open()

        ' Create the Command object
        Dim command As New SqlCommand()
        command.Connection = connection

        ' Create a string that defines the SQL statement
        Dim sql As String =
            "UPDATE Vendors " &
            "SET VendorState = UPPER(VendorState) " &
            "WHERE VendorID IN (SELECT VendorID FROM Inserted)"
        command.CommandText = sql

        ' Execute the Command object
        command.ExecuteNonQuery()

        ' Close the connection
        connection.Close()
    End Sub
End Class

File: SQL Server 2012/Projects/C#/Chapter 23/ApClrObjects/ApClrObjects/Triggers/Vendors_Update_State.cs
//------------------------------------------------------------------------------
// <copyright file="CSSqlTrigger.cs" company="Microsoft">
//     Copyright (c) Microsoft Corporation.  All rights reserved.
// </copyright>
//------------------------------------------------------------------------------
using System;
using System.Data;
using System.Data.SqlClient;
using Microsoft.SqlServer.Server;

public partial class Triggers
{        
    [Microsoft.SqlServer.Server.SqlTrigger (Target="Vendors", Event="AFTER INSERT, UPDATE")]
    public static void Vendors_Update_State ()
    {
        // open the Connection object for the context
        SqlConnection connection =
            new SqlConnection("Context connection=true");
        connection.Open();

        // create the Command object
        SqlCommand command = new SqlCommand();
        command.Connection = connection;

        // create a string that defines the sql statement
        string sql =
            "UPDATE Vendors " +
            "SET VendorState = UPPER(VendorState) " +
            "WHERE VendorID IN (SELECT VendorID FROM Inserted)";
        command.CommandText = sql;

        // execute the Command object
        command.ExecuteNonQuery();

        // close the Connection object
        connection.Close();
    }
}

